### PR TITLE
Remove `num256`

### PR DESCRIPTION
**Pull request by AaronKutch**

### CODE REVIEW

File: orchestrator/gravity_utils/Cargo.toml [hunk 1]
```diff
@@ -5,20 +5,18 @@ name = "gravity_utils"
 version = "0.1.0"
 
 [dependencies]
+clarity = { git = "https://github.com/AaronKutch/clarity.git", rev = "c50d7dca265d0b29da2326cae1b2996dc27b5ee1" }
```
[the remark this answers is not in the document]
Comment (AaronKutch) [reply orchestrator/gravity_utils/Cargo.toml:8]: ok

File: orchestrator/gbt/src/client/cosmos_to_eth.rs [hunk 2]
```diff
@@ -68,7 +68,8 @@ pub async fn cosmos_to_eth(
 
     match balance {
         Some(balance) => {
-            if balance.amount < amount.amount.clone() + bridge_fee.amount.clone() {
+            // TODO fix this `unwrap` along with the `exit`
```
[the remark this answers is not in the document]
Comment (AaronKutch) [reply orchestrator/gbt/src/client/cosmos_to_eth.rs:71]: Failures from the checked add operation need to be handled gracefully in conjunction with the `exit` when we fix issue #37 .

**Comment by AaronKutch** [comment]
I figured out that the old `num` dependency is being pulled in by `serde-rlp` which is itself 3 years since its last update. I might just leave it for now since it is conversely very stable.

File: orchestrator/gbt/src/client/cosmos_to_eth.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dzmitryhil) [reply orchestrator/gbt/src/client/cosmos_to_eth.rs:71]: I've fixed the #37 here the PR. https://github.com/onomyprotocol/cosmos-gravity-bridge/pull/59
So it would be better if you could fix this `unwrap` here in that PR.

**Comment by dzmitryhil** [comment]
(Resolves #44)

**Comment by AaronKutch** [comment]
resolves #41 also

File: orchestrator/cosmos_gravity/src/send.rs [hunk 3]
```diff
@@ -325,7 +321,14 @@ pub async fn send_to_eth(
     let mut found = false;
     for balance in balances {
         if balance.denom == amount.denom {
-            let total_amount = amount.amount.clone() + (fee.amount.clone() * 2u8.into());
+            let total_amount = amount
```
Comment (leviathanbeak) [inline orchestrator/cosmos_gravity/src/send.rs:324]: I like this :)

File: orchestrator/ethereum_gravity/src/logic_call.rs [hunk 4]
```diff
@@ -52,7 +54,7 @@ pub async fn send_eth_logic_call(
             before_nonce
         );
         return Ok(());
-    } else if current_block_height > call.timeout.into() {
+    } else if current_block_height > Uint256::from_u64(call.timeout) {
```
Comment (leviathanbeak) [inline orchestrator/ethereum_gravity/src/logic_call.rs:57]: any specific reason why we don't provide `into()` method for Uint256?

Comment (AaronKutch) [reply orchestrator/ethereum_gravity/src/logic_call.rs:57]: IIRC I did it because `uint` did not, and in Rust you cannot implement foreign traits for foreign types. In most cases `into` for integer casting does not save many characters because you need to specify `: Uint256` and add a `uX` suffix to the literal so Rust can infer both sides. `from_uX` is also more explicit.

File: orchestrator/gravity_utils/src/num_conversion.rs [hunk 5]
```diff
@@ -1,56 +1,8 @@
...
-        }
-        let mut lower_bytes: [u8; 16] = [0; 16];
-        // get the 'lowest' 16 bytes from a 256 bit integer
-        lower_bytes.copy_from_slice(&val[0..val.len()]);
-        Some(u128::from_be_bytes(lower_bytes))
-    }
-}
+const ONE_ETH_FLOAT: f64 = 1000000000000000000.;
+const ONE_GWEI_FLOAT: f64 = 1000000000.;
+const ONE_NOM_FLOAT: f64 = 1000000.;
```
Comment (dzmitryhil) [inline orchestrator/gravity_utils/src/num_conversion.rs:5]: Why the value for ONE_NOM is not "1000000000000000000" as it was before?

File: orchestrator/test_runner/src/main.rs [hunk 6]
```diff
@@ -75,9 +76,9 @@ lazy_static! {
 /// this value reflects the contents of /tests/container-scripts/setup-validator.sh
 /// and is used to compute if a stake change is big enough to trigger a validator set
 /// update since we want to make several such changes intentionally
-pub const STAKE_SUPPLY_PER_VALIDATOR: u128 = 1000000000000000000000;
+pub const STAKE_SUPPLY_PER_VALIDATOR: Uint256 = u256!(1000000000000000000000);
 /// this is the amount each validator bonds at startup
-pub const STARTING_STAKE_PER_VALIDATOR: u128 = STAKE_SUPPLY_PER_VALIDATOR / 2;
+pub const STARTING_STAKE_PER_VALIDATOR: Uint256 = u256!(500000000000000000000);
```
Comment (dzmitryhil) [inline orchestrator/test_runner/src/main.rs:81]: Why did you change to the fixed constant value instead of half of STAKE_SUPPLY_PER_VALIDATOR?

File: orchestrator/gravity_utils/src/num_conversion.rs (hunk 5 above)
Comment (AaronKutch) [reply orchestrator/gravity_utils/src/num_conversion.rs]: I must have merged the wrong way, I will fix it

**Comment by AaronKutch** [comment]
I ran cargo update one more time, hopefully will fix cargo audit

File: orchestrator/ethereum_gravity/src/logic_call.rs (hunk 4 above)
Comment (leviathanbeak) [reply orchestrator/ethereum_gravity/src/logic_call.rs:57]: yes I preferer being explicit but just noting that nothing is stopping you from implementing From<u64> and Into<u64> for your own Uint256 type

**Review by dzmitryhil — APPROVED** [review]
